Sources/SKCore/BuildSystemManager.swift

```diff
@@ -154,6 +155,37 @@ extension BuildSystemManager {
       .first
   }
 
+  /// Returns the target's module name as parsed from the `ConfiguredTarget`'s compiler arguments.
+  public func moduleName(for document: DocumentURI, in target: ConfiguredTarget) async -> String? {
```

Comment:
This feels like something that should be provided by the build system, rather than found from the settings after the fact. We could have a default that does this though.

It's also the last argument that wins, so both the `.first*` below should be `.last*` instead.

Reply:
Its my understanding that the build settings already give us a standardized way of getting the compilerArguments across all the difference build systems, so pushing the implementation down into each individual BuildSystem would be largely duplicating the implementation of `BuildSystem.buildSettings(for:in:language:)`. Let me know if this isn't what you meant.

Great catch on argument precedence, I'll put up a patch that fixes it on main and then roll it in to this cherry-pick.

Reply:
I've created https://github.com/swiftlang/sourcekit-lsp/pull/1549 to fix argument precedence

Reply:
> Its my understanding that the build settings already give us a standardized way of getting the compilerArguments across all the difference build systems

This is true, but doesn't necessarily mean we should go searching through build settings for the information we want (I would consider that a fallback) - the build system may already have the information needed. And like I said, we could always have a default implementation for that method that does as you are now.